core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -64,7 +64,10 @@ async def auto_restart_ardupilot(self) -> None:
                 self.ardupilot_subprocess is not None and self.ardupilot_subprocess.poll() is not None
             ) or len(self.running_ardupilot_processes()) == 0
             needs_restart = self.should_be_running and (
-                (self.current_platform in [Platform.SITL, Platform.Navigator] and process_not_running)
+                (
+                    self.current_platform in [Platform.SITL, Platform.Navigator, Platform.NavigatorR3]
```

Comment:
I'm not sure if I follow, why Navigator R3 is here if it follows the same string as Navigator ?

Reply:
Because the only specific enum is Platform, and NavigatorR4 is a different platform from NavigatorR3 (they happen to have the same string, but comparing strings gets unnecessarily unsafe).

We can keep just a Platform.NavigatorR5 there, but if we happen to change the strings for Navigator platforms (R3, R4, etc) that won't be tracked, and we may miss something.

Reply:
I'm thinking adding an extra field like `type`, which could be Serial or Linux would help to get this less verbose (specially when we get to support more boards).